### PR TITLE
Test module

### PR DESCRIPTION
**Pull request by amartyanambiar**
Test Cases for adding trees between nodes added.

### Instructions :
1. Always start from `basic_science` node to `ganesha_statue` and complete the loop.

### CODE REVIEW

File: test_module/test.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            self.G.add_tree(tc.tree10.point_coordinate, start_node, end_node)
+            self.G.save_and_return()
+
+        elif self.start_node == 'teacher_parking' and self.end_node == 'basic_science':
+            self.G.add_tree(tc.tree11.point_coordinate, start_node, end_node)
+            self.G.add_tree(tc.tree12.point_coordinate, start_node, end_node)
+            self.G.save_and_return()
+
+        else:
+            pass
```
Comment (amartyanambiar) [inline test_module/test.py:41]: Note that the conditions are based on the nodes in a loop. from basic_science and so on.

order to add trees on interface would be :
basic_science <-> ganesha_statue
ganesha_statue <-> hostel_turn
hostel_turn <-> volley_ball_court
volley_ball_court  <->  teacher_parking
teacher_parking  <->  teacher_parking

File: app.py [hunk 2]
```diff
@@ -63,80 +66,31 @@ def automate_vehicle():
...
+@app.route("/add_tree", methods=["POST"])
 def tree():
-    gt.add_tree([12.96620, 77.71203])
-    gt.add_tree([12.96612, 77.71179])
-    gt.add_tree([12.96603, 77.71156])
-    return "a"
+    start_node = request.get_json().get("start_node")
+    end_node = request.get_json().get("end_node")
+    TestTree(gt, start_node, end_node)
+    return 'a'
```
Comment (amartyanambiar) [inline app.py:74]: Changed to retrieve the selected nodes by the user on interface.

File: static/tree.js [hunk 3]
```diff
@@ -1,46 +1,55 @@
...
 
-addTreeButton.addEventListener('click', () => {
-  fetch('/add_tree')
+  fetch("/add_tree", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({
+      start_node: selectedStart,
+      end_node: selectedEnd,
+    }),
```
Comment (amartyanambiar) [inline static/tree.js:51]: Made changes to post the start and end nodes of the stretch to backend

File: test_module/test_coordinates.py [hunk 4]
```diff
@@ -18,14 +18,19 @@
...
+# V
+tree10 = Coordinates("tree10", [12.96700, 77.71190])
+tree11 = Coordinates("tree11", [12.96692, 77.71192])
+# teacher_parking
+teacher_parking = Coordinates("teacher_parking", [12.9667974, 77.711963])
+# |
+# V
+tree11 = Coordinates("tree11", [12.96659, 77.71202])
+tree12 = Coordinates("tree12", [12.96645, 77.71206])
+# to basic_science
```
Comment (amartyanambiar) [inline test_module/test_coordinates.py]: Error resolved . duplicate values.

and added two trees between volley_ball_court and teacher_parking and basic_science

File: unpickle.py [hunk 5]
```diff
@@ -7,6 +7,10 @@
...
-shortest_route = nx.shortest_path(
-    gt_loaded, 't-1', 'volley_ball_court', weight='length')
-print(shortest_route)
+# shortest_route = nx.shortest_path(
+#     gt_loaded, 't-1', 'volley_ball_court', weight='length')
+# print(shortest_route)
+
+fig, ax = ox.plot_graph(gt_loaded, show=False, close=False)
+fig.savefig('output/graph.png', dpi=300,
+            bbox_inches='tight', pad_inches=0)
```
Comment (amartyanambiar) [inline unpickle.py:16]: plotting the graph of the created .pkl file

**Review by Blake2912 — APPROVED** [review]
Go for merge... you can integrate the same with the recent code pushes in `simulation-main`. Will create a issue describing the requirements

**Pull request merged**